packages/discord.js/src/structures/Attachment.js

```diff
@@ -3,74 +3,30 @@
 const Util = require('../util/Util');
 
 /**
- * Represents an attachment.
+ * @typedef {Object} AttachmentPayload
+ * @property {?string} name
+ * @property {Stream|BufferResolvable} attachment
+ * @property {?string} description
+ */
+
+/**
+ * Represents an attachment
  */
 class Attachment {
   /**
-   * @param {BufferResolvable|Stream} attachment The file
-   * @param {string} [name=null] The name of the file, if any
-   * @param {APIAttachment} [data] Extra data
+   * @param {APIAttachment} [data] Attachment data
+   * @private
    */
-  constructor(attachment, name = null, data) {
-    this.attachment = attachment;
+  constructor(data) {
+    this.attachment = data.attachment;
     /**
      * The name of this attachment
      * @type {?string}
      */
-    this.name = name;
+    this.name = data.name;
```

Comment:
`APIAttachment`s don't have an `attachment` property, and don't have a `name` property, it's `filename`, and it's not nullable
https://github.com/discordjs/discord-api-types/blob/main/payloads/v10/channel.ts#L1035-L1078